js/main.js

```diff
@@ -148,7 +148,31 @@ document.addEventListener('click', function (event) {
   }
   if (event.target.matches('a.edit')) {
     viewSwapping('edit-profile');
-  } else if (event.target.matches('a.link') && data.profile.username !== '') {
+  } else if (event.target.matches('a.link.prof') && data.profile.username !== '') {
```

Comment:
Instead of checking if `data.profile.username` is not empty on each condition here, you could refine this by checking if the event target matches `a.edit` OR `data.profile.username` is an empty string. By doing so, if you get to any subsequent conditional blocks, you can be sure that there is a value in the username property :+1: